kybra/compiler/kybra_generate/src/get_name.rs

```diff
@@ -1,32 +1,44 @@
 use rustpython_parser::ast::{ExprKind, Located, StmtKind};
 
-use crate::source_map::SourceMapped;
+use crate::{
+    candid_type::errors::{InvalidName, NotExactlyOneTarget},
+    source_map::SourceMapped,
+    Error,
+};
 
 impl SourceMapped<&Located<StmtKind>> {
-    pub fn get_name(&self) -> Option<String> {
-        match &self.node {
-            StmtKind::FunctionDef { name, .. } => Some(name.clone()),
-            StmtKind::AsyncFunctionDef { name, .. } => Some(name.clone()),
-            StmtKind::ClassDef { name, .. } => Some(name.clone()),
+    pub fn get_name(&self) -> Result<Option<&str>, Error> {
+        Ok(match &self.node {
+            StmtKind::FunctionDef { name, .. } => Some(name),
+            StmtKind::AsyncFunctionDef { name, .. } => Some(name),
+            StmtKind::ClassDef { name, .. } => Some(name),
             StmtKind::Assign { targets, .. } => {
                 if targets.len() != 1 {
-                    None
-                } else {
-                    SourceMapped::new(&targets[0], self.source_map.clone()).get_name()
+                    return Err(NotExactlyOneTarget::err_from_stmt(self).into());
```

Comment:
Double check this is good. Target might not be clear enough for a end-developer.